### PR TITLE
Replace returnValue with return

### PR DESCRIPTION
**Pull request by hawkowl**
It makes the tracebacks a _little_ better, but still leaks info in logs. It'll help for when we move to `async`/`await`, anyway.

### CODE REVIEW

File: synapse/federation/federation_server.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def _handle_incoming_transaction(self, origin, transaction, request_time):
                 "[%s] We've already responded to this request",
                 transaction.transaction_id,
             )
-            defer.returnValue(response)
+            return response
             return
```
Comment (richvdh) [inline synapse/federation/federation_server.py]: lolz

File: scripts-dev/strip_returnvalue.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import re
```
Comment (richvdh) [inline scripts-dev/strip_returnvalue.py]: this doesn't feel worth committing?

Comment (hawkowl) [reply scripts-dev/strip_returnvalue.py]: if some later merges have code using it i can just scrounge it from my hdd I guess :)

File: synapse/federation/federation_server.py (hunk 1 above)
Comment (hawkowl) [reply synapse/federation/federation_server.py]: lmao

**Pull request merged**